crypto3/libs/math/include/nil/crypto3/math/polynomial/polynomial_dfs.hpp

```diff
@@ -605,6 +605,23 @@ namespace nil {
                     return polynomial_dfs(new_s - 1, r);
                 }
 
+                /* Inverses the values in polynomial using Montgomery trick.
+                 * Calls inverse on a group element just once, so it's much faster than inverting each element separately.
+                 */
+                void inverse() {
```

Comment:
Maybe rename it somehow?
Result is not reversion of the polynomial.
If we multiply polynomial on .inverse() result it will be not 1 polynomial but higher degree polynomial with 1-s on domain elements.

Reply:
Renamed to element_wise_inverse